charts/cortex/templates/store-gateway/store-gateway-statefulset.yaml

```diff
@@ -129,4 +130,152 @@ spec:
+{{- range $availabilityCell := .Values.topologyAware.availabilityCells }}
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: {{ template "cortex-store-gateway.fullname" $ }}-{{ $availabilityCell }}
+  labels: {{ include "cortex-store-gateway.labels" $ | nindent 4 }}
+{{- $.Values.topologyAware.zoneLabel | nindent 4 }}: {{ $availabilityCell }}
+  annotations:
+    {{- toYaml $.Values.store_gateway.annotations | nindent 4 }}
+spec:
+  replicas: {{ $.Values.store_gateway.replicas }}
+  selector:
+    matchLabels: {{ include "cortex-store-gateway.matchLabels" $ | nindent 6 }}
+{{- $.Values.topologyAware.zoneLabel | nindent 6 }}: {{ $availabilityCell }}
+  updateStrategy:
+    {{- toYaml $.Values.store_gateway.strategy | nindent 4 }}
+  serviceName: {{ template "cortex-store-gateway.headlessname" $ }}
+  {{- if $.Values.store_gateway.persistentVolume.enabled }}
+  volumeClaimTemplates:
+    - metadata:
+        name: storage
+        {{- if $.Values.store_gateway.persistentVolume.annotations }}
+        annotations:
+          {{ toYaml $.Values.store_gateway.persistentVolume.annotations | nindent 10 }}
+        {{- end }}
+      spec:
+        {{- if $.Values.store_gateway.persistentVolume.storageClass }}
+        {{- if (eq "-" $.Values.store_gateway.persistentVolume.storageClass) }}
+        storageClassName: ""
+        {{- else }}
+        storageClassName: "{{ $.Values.store_gateway.persistentVolume.storageClass }}"
+        {{- end }}
+        {{- end }}
+        accessModes:
+          {{- toYaml $.Values.store_gateway.persistentVolume.accessModes | nindent 10 }}
+        resources:
+          requests:
+            storage: "{{ $.Values.store_gateway.persistentVolume.size }}"
+  {{- end }}
+  template:
+    metadata:
+      labels: {{ include "cortex-store-gateway.labels" $ | nindent 8 }}
+{{- $.Values.topologyAware.zoneLabel | nindent 8 }}: {{ $availabilityCell }}
+        {{- with $.Values.store_gateway.podLabels }}
+        {{- toYaml . | nindent 8 }}
+        {{- end }}
+      annotations:
+        checksum/config: {{ include "cortex.configChecksum" $ }}
+        {{- with $.Values.store_gateway.podAnnotations }}
+        {{- toYaml . | nindent 8 }}
+        {{- end }}
+    spec:
+      serviceAccountName: {{ $.Values.store_gateway.serviceAccount.name | default (include "cortex.serviceAccountName" $ ) }}
+      {{- if $.Values.store_gateway.priorityClassName }}
+      priorityClassName: {{ $.Values.store_gateway.priorityClassName }}
+      {{- end }}
+      securityContext: {{- toYaml $.Values.store_gateway.securityContext | nindent 8 }}
+      initContainers:
+        {{- toYaml $.Values.store_gateway.initContainers | nindent 8 }}
+      {{- if $.Values.image.pullSecrets }}
+      imagePullSecrets:
+      {{- range $.Values.image.pullSecrets }}
+        - name: {{ . }}
+      {{- end }}
+      {{- end }}
+      nodeSelector:
+        {{- toYaml $.Values.store_gateway.nodeSelector | nindent 8 }}
```

Comment:
I think node selector need to be part of the of `topologyAware.availabilityCell` configuration structure as it has to select a node from a different zone for each statefulset.

Reply:
I think you removed the node selector for the ingester probably we want to be consistent here?